lib/src/library_scanner.dart

```diff
@@ -40,40 +38,34 @@ class LibraryScanner {
   LibraryScanner._(
       this.packageName, this._packagePath, this._session, this._overrides);
 
-  factory LibraryScanner(String dartSdkPath, String packagePath,
-      {List<LibraryOverride> overrides}) {
+  static Future<LibraryScanner> create(String dartSdkPath, String packagePath,
+      {List<LibraryOverride> overrides}) async {
```

Comment:
Wouldn't it be safer to pass in a `packageName` to be analyzed instead of `packagePath`? Or read the package name from `pubspec.yaml`..

Reply:
We can also leave a comment to cleanup this logic.

Reply:
Yeah, that would make much more sense, I don't remember why we have it this way...